Feat/#92 메뉴 단일 조회 ,  메뉴 API 개선

uzjaee opened the pull request:
## #️⃣ 해당 변경 사항과 연관된 이슈는 무엇인가요?

>  #92  #93

## ✏️ 작업한 내용을 간략히 설명해 주세요!
단일 메뉴 조회 기능 구현 
메뉴 api 형태 수정 
- endpoint : menu >menus
- response 형태 통일 
- 메뉴 생성 api location 값 올바른 api 로 수정 
- 메뉴 API 개선 

## 🛠️ 변경을 위해 진행한 작업 내용에는 어떤 것이 있나요?

- 📌 cotroller ,service ,repo 코드 추가 
- 📌 테스트 코드 작성 
- 📌 audit 컬럼 적용 
- 📌 exception 적용     

## 📝 변경 사항을 확인하기 위해 테스트한 방법을 설명해 주세요.

>  http client 

## ✅ 해당 기능에 대한 테스트가 작성되었나요?

- [ ] 🟢 작성
- [x ] 🟡 일부 구현
- [ ] 🔴 미 작성

## 💬 리뷰 요구사항

> 같이 논의했으면 하는 사항이 있으신가요?

## 📚 참고 자료

> 구현에 참고한 자료가 있다면 공유해주세요!

---

## 종료할 이슈는 무엇인가요?

close #92 close #93

uzjaee committed:
feat(#92) 메뉴 단일 조회 API 구현
FindAllMenuResponse.java > FindMenuResponse.java 로이름 변경 ( dto의 역할과 이름이 맞지 않았음)
controller , service , repo 내 메뉴 단일 조회 구현

uzjaee committed:
refactoring(#93) 메뉴 API 리팩토링
메뉴 API endPoint : menu > menus 변경
http client 내 테스트 api endpoint 수정
SecurityConfig 내 허용 경로 수정
menu-create-api.http 삭제

uzjaee committed:
feat(#92) 메뉴 단일 조회 API 테스트 코드 구현

uzjaee committed:
refactor(#93) 메뉴 API response 및 location
각 api response 형태 통일
메뉴 생성 api Location 올바른 주소로 변경

eggnee commented on src/main/java/com/sparta/baedallegend/domains/menu/controller/MenuController.java:
```diff
@@ -14,33 +14,42 @@
+	@GetMapping("/menu/{menuId}")
+	public ResponseEntity<FindMenuResponse> FindOneMenu(
+		@PathVariable String menuId) {
+		FindMenuResponse memo = menuService.findOneMenu(menuId);
```
왜 변수명이 memo 죠?! 😏

uzjaee replied:
게슈탈트 붕괴현상인가요 .. 왜 메모죠 ? ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ
수정 반영할게요 !

eggnee commented on src/main/java/com/sparta/baedallegend/domains/menu/repo/MenuRepo.java:
```diff
@@ -17,7 +18,10 @@ public interface MenuRepo extends JpaRepository<Menu, UUID> {
 
 	List<Menu> findByShopAndIsPublicTrue(Shop shop);
 
+	Optional<Menu> findByIdAndIsPublicTrue(UUID menuId);
```
`@SQLRestriction(DEFAULT_CONDITION)` 을 활용해보는 거 어떠신가요!!!
`CommonAuditFields` 을 참고하시면 됩니다!

eggnee commented on src/main/java/com/sparta/baedallegend/domains/menu/service/MenuService.java:
```diff
@@ -32,15 +32,22 @@ public String create(CreateMenuRequest createMenuRequest) {
+
+	public FindMenuResponse findOneMenu(String menuId) {
+		Menu menu = menuRepo.findByIdAndIsPublicTrue(UUID.fromString(menuId)).orElseThrow(
+			() -> new IllegalArgumentException("존재하지 않는 메뉴입니다.")
```
용재님도 `MenuException`을 따로 구현해보는 거 어떠신가요!

uzjaee committed:
refactor(#93) menu Entity Auditable 적용
menu entity 내 코드 SQLRestriction 적용
menuRepo ispublic 과 관련된 것 코드 수정

uzjaee committed:
chore(#93) shop_id > shopId로 변경

uzjaee committed:
feat(#93): 메뉴 Exception 추가
메뉴 Exception 코드 구현
enum MenuErrorCode 구현 및 NOT_EXIST추가
메뉴서비스 내 shopException , menuexception사용
메뉴 서비스 내 pagable 객체 content가 비어있는지 확인하여 exception 사용
메뉴 API를 통한 테스트

uzjaee committed:
Merge branch 'develop' into feat/#92-find-one-menu

uzjaee commented:
audit Fields 사용을 통한 jpa 변경 완료
menu Exception 구현 및 사용을 통한 예외 처리 완료
